### PR TITLE
Add port_numbers API

### PR DESCRIPTION
**Pull request by serpilliere**
Add `port_numbers` API.
The result of #41 maybe taken into account before.

### CODE REVIEW

File: src/device.rs [hunk 1]
```diff
@@ -139,4 +140,31 @@ impl<T: UsbContext> Device<T> {
...
+        let ports_number = if result < 0 {
+            return Err(error::from_libusb(result));
+        } else {
+            result
+        };
+        if ports_number as usize > ports.len() {
+            return Err(error::Error::Overflow);
+        }
+
+        ports.drain(0..ports.len() - ports_number as usize);
```
Comment (a1ien) [inline src/device.rs:166]: Are you sure it's that what you want?
if libusb_get_port_numbers return 4 and ports vec contain vec![1, 2, 3, 4, 0, 0, 0];
after call `ports.drain(0..ports.len() - 4 as usize);` we have ports = [  0,  0,  0,]

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=53785a81342c55d2ce0e4644d0c580d8

Comment (a1ien) [reply src/device.rs:166]: Maybe `ports.drain(ports_number..);`

Comment (serpilliere) [reply src/device.rs:166]: You are right. I pushed the wrong fix: 
I have tested on a simple example which gave me in the original buffer:
```
[2020-10-07T16:32:42Z INFO  test_usb_parent] Device 1 22
[2020-10-07T16:32:42Z INFO  test_usb_parent] Found!
xxx [6, 3, 0, 0, 0, 6, 3]
[2020-10-07T16:32:42Z INFO  test_usb_parent] Found Ok([6, 3])
```
I took the last part of the buffer, but the libusb code clearly does the memmove to put the data at the beginning of the buffer.
Sorry for this.

**Comment by serpilliere** [comment]
I have updated the fix

**Comment by serpilliere** [comment]
Hi!
I think I took all your comments into account.

**Comment by a1ien** [comment]
Thanks!

**Comment by serpilliere** [comment]
Thanks to you @a1ien !

**Pull request merged**